packages/components/src/draggable/index.tsx

```diff
@@ -207,14 +207,8 @@ export function Draggable( {
 		// Update cursor to 'grabbing', document wide.
 		ownerDocument.body.classList.add( bodyClass );
 
-		// Allow the Synthetic Event to be accessed from asynchronous code.
-		// https://reactjs.org/docs/events.html#event-pooling
-		event.persist();
-
-		let timerId: number | undefined;
-
 		if ( onDragStart ) {
-			timerId = setTimeout( () => onDragStart( event ) );
+			onDragStart( event );
```

Comment:
Synchronously passing the `event` to the callback is essential for the consumers to be able to call `event.dataTransfer.setData`.

Reply:
TBH, I'm not sure why we even wanted to defer the callback of `onDragStart`, all other event callbacks seem to be called synchronously. This is more of a _correct_ change that also conveniently enables us to set multiple types of `dataTransfer` in the event callback (and other components). I'd say this is more like a bug fix because currently accessing the `dataTransfer` in `onDragStart` causes unexpected behavior and doesn't follow the spec.

Your solution works too and might be a better solution if we want to minimize the impact of such PR! However, both solution have breaking changes, we just have to decide which is the preferred one at this stage :).

Reply:
I don't see how my PR has breaking changes? 🤔

Reply:
Correct me if I'm wrong, `<Draggable>` is an exported component of the package, so changing the way `dataTransfer` works for strings is going to be a breaking change for anyone consuming the package.

Reply:
Hm... you're right. It could be addressed though by just adding a flag to stringify the data or not([commit](https://github.com/WordPress/gutenberg/pull/49297/commits/003f774e79b8d15b828617a7b37684dfd80c0fcc))

Reply:
Personally I don't think adding a prop to fix a backward-compatibility issue which we'll have to support for backward-compatibility is going to worth it 😅. I think making `onDragStart` synchronous is the more _correct_ fix in the long run and will likely fix some other issues with the component. Currently, consumers using this component will not be able to set multiple data types because of this _bug_. Fixing it hopefully will make it more useful for them. This is just my opinion though, and I'll respect your and others' opinions if we decide to go with your approach!